tests/lib/c_lib/src/main.c

```diff
@@ -612,6 +699,103 @@ void test_strtok_r(void)
 	test_strtok_r_do("1|2|3,4|5",           "| ", 5, tc01, false);
 }
 
+/**
+ *
+ * @brief Test snprintf function
+ *
+ */
+
+void test_snprintf(void)
+{
+	int ret;
+	char str[10];
+
+	ret = snprintf(str, 3,  "%s", "abcd");
+	zassert_equal(ret, 4, "snprintf 3");
+	ret = snprintf(str, 0,  "%s", "abcd");
+	zassert_equal(ret, 4, "snprintf 3");
+}
+
+/**
+ *
+ * @brief Test fprintf function
+ *
+ */
+
+void test_fprintf(void)
+{
+	int ret, i = 3;
+
+	ret = fprintf(stdout,  "%d", i);
+	zassert_equal(ret, 1, "fprintf 3");
+}
+
+/**
+ *
+ * @brief Test vfprintf function
+ *
+ */
+
+int WriteFrmtd_vf(FILE *stream, char *format, ...)
+{
+	int ret;
+	va_list args;
+
+	va_start(args, format);
+	ret = vfprintf(stream, format, args);
+	va_end(args);
+
+	return ret;
+}
+
+void test_vfprintf(void)
+{
+	int ret;
+
+	ret = WriteFrmtd_vf(stdout,  "This %d", 3);
+	zassert_equal(ret, 6, "vfprintf This 3");
+}
+
+/**
+ *
+ * @brief Test vprintf function
+ *
+ */
+
+int WriteFrmtd_v(char *format, ...)
+{
+	int ret;
+	va_list args;
+
+	va_start(args, format);
+	ret = vprintf(format, args);
+	va_end(args);
+
+	return ret;
+}
+
+void test_vprintf(void)
+{
+	int ret;
+
+	ret = WriteFrmtd_v("This %d", 3);
+	zassert_equal(ret, 6, "vprintf This 3");
+}
+
+/**
+ *
+ * @brief Test fputs function
+ *
+ */
+
+void test_fputs(void)
+{
+	int ret;
+
+	ret = fputs("This 3", stdout);
```

Comment:
I think we test fputs() with null pointer.

Reply:
Yes, It works.

Reply:
It didn't  allow to pass NULL as a parameter to fputs function due to the compile error in some platforms, so change the null to stdin which works too.